### ISSUE TITLE
[Question]: Custom contract resolver for properties with private setters being "ignored" by JsonSerialiser.

### ISSUE THREAD

**Issue by explorer14** [aspnet/Mvc#6424]
Hi,

This is not an issue, just a question I am seeking some help on.

I have a Customer class with the Id property having a private setter for good reason:

`public class Customer`
`{`
    `private Guid id;`
    `private string name;`
    `public Guid Id { get => id; private set => id = value; }`
    `public string Name { get => name; set => name = value; }`
`}`

I am updating an existing Customer record in my view using ajax form postback which uses json serialisation/deserialisation to send the data to the controller action including the Id in a hidden field.

On my controller side, I expect the JsonDeserialiser to be able to assign to the Id property when I am doing an edit operation. In the case of public setters this works, but Id has a private setter so for this, I am trying to assign a custom contract resolver during startup that allows writing to properties with private setters but for some reason it seems that the resolver is being ignored because neither the breakpoint in the resolver gets hit at the point of deserialisation nor the Id property on the object is hydrated (just shows Guid.Empty) or may be I am doing something wrong, I don't know! Can anyone help please? 

My ConfigureServices method in Startup.cs (these calls to `AddMvc()` and `AddJsonOptions()` are the first things in this method before any other services being registered):

`services.AddMvc()`
`                    .AddJsonOptions(options => `
`                    {`
`                        options.SerializerSettings.ContractResolver = new ReadonlyPropertyContractResolver();`
`                    });`

My HomeController::Create action (I am using the same action for creating and editing a customer):

`[HttpPost]`
`        public IActionResult Create(Customer customer)`
`        {`
`            this.customerRepo.AddOrUpdate(customer);`
`                return RedirectToAction("Index");`
`        }`

Any help with this will be greatly appreciated and please do let me know if you need any more info related to this.

Thanks
Aman

**Comment by rynowak** [comment]
Why is this closed? did you find what you are looking for?

**Comment by explorer14** [comment]
Sorry, I closed this because I realised the silliness of my question :). MVC model binding has got nothing to do with JSON serialisation/deserialisation its a completely different process, if I am not wrong. The resolver wasn't being ignored, it just wasn't what the model binding uses for sending data back and forth. WebAPI projects on the other hand usually send and recieve JSON and that's where a contract resolver is applicable. 

So the question was totally unrelated to what was actually happening, misinterpretation on my part. I simplified my code a bit and solved the problem a different a bit more generic way. Thanks for asking!

Cheers
Aman
